packages/mdc-slider/_mixins.scss

```diff
@@ -0,0 +1,105 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "@material/theme/mixins";
+@import "./variables";
+
+//
+// Public
+//
+
+@mixin mdc-slider-rail-color($color) {
+  @include mdc-slider-rail-color-impl_($color);
+}
+
+@mixin mdc-slider-thumb-color($color) {
+  @include mdc-slider-thumb-color-impl_($color);
+}
+
+@mixin mdc-slider-focus-halo-color($color) {
+  .mdc-slider__focus-ring {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-slider-value-pin-ink-color($color) {
+  .mdc-slider__pin {
+    @include mdc-theme-prop(color, $color);
+  }
+}
+
+@mixin mdc-slider-value-pin-fill-color($color) {
+  .mdc-slider__pin {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-slider-value-pin-fill-color-accessible($color) {
+  $ink-color: if(mdc-theme-tone($color) == "light", text-primary-on-light, text-primary-on-dark);
+
+  @include mdc-slider-value-pin-fill-color($color);
+  @include mdc-slider-value-pin-ink-color($ink-color);
+}
+
+// NOTE: This mixin sets the color of ALL customizable elements in the slider. If new elements are added, this mixin
+// should be updated to cover those additional elements.
+@mixin mdc-slider-color-accessible($color) {
+  @include mdc-slider-rail-color($color);
+  @include mdc-slider-thumb-color($color);
+  @include mdc-slider-focus-halo-color($color);
+  @include mdc-slider-value-pin-fill-color-accessible($color);
+}
+
+//
+// Private
+//
+
+@mixin mdc-slider-rail-color-impl_($color, $important: false) {
+  $rail-color: rgba(mdc-theme-prop-value($color), .26);
+
+  // Rail
+  .mdc-slider__track-container {
+    @include mdc-theme-prop(background-color, $rail-color, $important);
+  }
+
+  // Highlight
+  .mdc-slider__track {
+    @include mdc-theme-prop(background-color, $color, $important);
+  }
+
+  // Tick marks
+  .mdc-slider__track-marker::after,
+  .mdc-slider__track-marker-container::after {
+    @include mdc-theme-prop(background-color, $color, $important);
+  }
+}
+
+@mixin mdc-slider-thumb-color-impl_($color, $important: false) {
+  .mdc-slider__thumb {
+    @include mdc-theme-prop(fill, $color, $important: $important);
+    @include mdc-theme-prop(stroke, $color, $important: $important);
+  }
+}
+
+@mixin mdc-slider-thumb-stroke-cutout_($color) {
+  // We use !important here to override all styles without having to use excessive
+  // :not() selectors or intricate compound selectors.
```

Comment:
What is "excessive :not() selectors"? How many :not selectors would you need to add?

Reply:
I actually just copied that comment from the original Sass code. At the time, I hadn't removed the `--off` modifier yet, so on the surface it seemed plausible that the comment was still accurate.

However, I just tried using `:not()` selectors, and it was actually super easy, so I updated the PR to do that instead of using `!important` at all.

Good catch!